lib/presentation/widgets/list_characters.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:marvel/bloc/heroes_bloc.dart';
+import 'package:marvel/bloc/heroes_state.dart';
+import 'package:marvel/data/model/character.dart';
+import 'package:marvel/presentation/navigation/app_routes.dart';
+
+class ListCharacters extends StatelessWidget {
+  const ListCharacters({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<HeroesBloc, HeroesState>(
+      builder: (context, state) {
+        if (state.loading == true) {
+          return const Center(
+            child: CircularProgressIndicator(),
+          );
+        }
+        if (state.loading == false && state.characters != null) {
+          return ListView.builder(
+            itemCount: state.characters?.length,
+            itemBuilder: (BuildContext context, int index) {
+              final item = state.characters![index];
+              return SizedBox(
+                height: 100,
+                child: GestureDetector(
+                  onTap: () {
+                    Navigator.pushNamed(context, AppRoutes.detailPage,
+                        arguments: Character(
```

Comment:
You should pass throught arguments `id` only and using this id on details page fetch the details.

Reply:
done